elmslie-core/src/main/kotlin/money/vivid/elmslie/core/store/Actor.kt

```diff
@@ -3,18 +3,37 @@ package money.vivid.elmslie.core.store
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import money.vivid.elmslie.core.switcher.Switcher
+import kotlin.reflect.KClass
 
 abstract class Actor<Command : Any, Event : Any> {
 
+    private val mutex = Mutex()
+    private val switchers = mutableMapOf<KClass<out Any>, Switcher>()
+
     /**
      * Executes a command. This method is performed on the [Dispatchers.IO]
      * [kotlinx.coroutines.Dispatchers.IO] which is set by ElmslieConfig.ioDispatchers()
      */
-    abstract fun execute(command: Command): Flow<Event>
+    abstract suspend fun execute(command: Command): Flow<Event>
```

Comment:
There is some 'breaking' change; for me it's more honest for users since they can do some work before making flow. 
Also we start collecting flow inside separate coroutine anyway..

Otherwise I can't make getting 'switcher' thread-safe